tests/kafkatest/tests/connect/connect_test.py

```diff
@@ -127,3 +128,73 @@ def validate_output(self, value):
+        self.enable_deadletterqueue = True
+
+        successful_records = []
+        faulty_records = []
+        all_records = []
+        for i in range(0, 1000):
+            if i % 2 == 0:
+                all_records.append('{"some_key":' + str(i) + '}')
+                successful_records.append('{some_key=' + str(i) + '}')
+            else:
+                # badly formatted json records (missing a quote after the key)
+                all_records.append('{"some_key:' + str(i) + '}')
+                faulty_records.append('{"some_key:' + str(i) + '}')
+        all_records = "\n".join(all_records) + "\n"
+        successful_records = "\n".join(successful_records) + "\n"
+        if error_tolerance == "all":
+            faulty_records = ",".join(faulty_records)
+        else:
+            faulty_records = faulty_records[0]
+
+        self.source = ConnectStandaloneService(self.test_context, self.kafka, [self.INPUT_FILE, self.OFFSETS_FILE])
+        self.sink = ConnectStandaloneService(self.test_context, self.kafka, [self.OUTPUT_FILE, self.OFFSETS_FILE])
+
+        self.zk.start()
+        self.kafka.start()
+
+        self.override_key_converter = "org.apache.kafka.connect.storage.StringConverter"
+        self.override_value_converter = "org.apache.kafka.connect.storage.StringConverter"
+        self.source.set_configs(lambda node: self.render("connect-standalone.properties", node=node), [self.render("connect-file-source.properties")])
+
+        self.override_key_converter = "org.apache.kafka.connect.json.JsonConverter"
+        self.override_value_converter = "org.apache.kafka.connect.json.JsonConverter"
+        self.override_key_converter_schemas_enable = False
+        self.override_value_converter_schemas_enable = False
+        self.sink.set_configs(lambda node: self.render("connect-standalone.properties", node=node), [self.render("connect-file-sink.properties")])
+
+        self.source.start()
+        self.sink.start()
+
+        # Generating data on the source node should generate new records and create new output on the sink node
+        self.source.node.account.ssh("echo -e -n " + repr(all_records) + " >> " + self.INPUT_FILE)
+
+        if error_tolerance == "none":
+            try:
+                wait_until(lambda: self.validate_output(successful_records), timeout_sec=15,
+                           err_msg="Clean records added to input file were not seen in the output file in a reasonable amount of time.")
+                raise Exception("Expected to not find any results in this file.")
+            except TimeoutError:
+                self.logger.info("Caught expected exception")
+        else:
+            wait_until(lambda: self.validate_output(successful_records), timeout_sec=15,
+                       err_msg="Clean records added to input file were not seen in the output file in a reasonable amount of time.")
+
+        self.logger.info("Reading records from deadletterqueue")
```

Comment:
probably inside the `if` instead of before since it's kind of confusing to see this for a test that isn't going to actually read the DLQ